### DISCUSSION TITLE
CustomTube's future fork

### DISCUSSION THREAD

**Discussion by Lort533** [lightbeam24/CustomTube#210] [General]
Hello,
I would like to ask, like the title says - is this project alive or abandoned? It's been a month since *any* activity on the repo. No commits? That's understandable, but there are no replies to new issues too. Is there a reason why does this happen? If the repo owner reads this right now, I'd just like to know if the project is abandoned, or just temporarily left aside, or maybe there is actually something happening "behind the scenes", new beta (or an actual release) that's simply not public yet. I may be a bit paranoid to ask such a question after just a month of inactivity, but it's somewhat rather a month already.

**Comment by MesaBlack** [comment]
???

**Comment by Aleksandra0056** [comment]
I agree, this project needs urgent fixes immediately because many YouTube layouts are broken or glitched.

**Comment by Lort533** [comment]
I can confirm that 2012 layout seems to slowly break apart, especially visible on playlists and livestreams.
Hear me out, I'm here to help, but I don't want to fork this project and extensively work on it if e.g. the project owner DOES care about the project, but doesn't have time, motivation, etc., I think you know what I mean, making an updated, improved alternative that will cause issues to owner of this project (of being behind these changes).

**Reply by Aleksandra0056** [reply]
I agree, we can wait 2 (or 3) weeks for any commits, and if no changes happen, then this project is probably abandoned.

For example, the project [Firefox Scripts](https://github.com/xiaoxiaoflood/firefox-scripts) (by xiaoxiaoflood) faced a similar fate, that project was often updated, but from one point of time, no changes were made at all making it abandoned.

EDIT: The last contribution made by lightbeam24 was on 19th March (but that was for a foreign project), so we still have a small hope, but I don't think, because I don't want to believe, that someone is ignoring important bug fixes for his own project.

**Comment by Lort533** [comment]
Alright, so it's been a while and I randomly came across this:
https://www.reddit.com/r/oldyoutubelayout/

Well, from reading what's in there it seems like CustomTube is __unofficially__ discontinued, and the project repo owner works on something else, thought this project is not abandoned as the author didn't decide yet.

Not sure why the repo owner is inactive here, as he is indeed active there on Reddit.

Apparently there is something else, a better replacement going on there. But I don't really wanna mess with installing 3rd party stuff, as an extension that is officially available on Google and Firefox web stores + GitHub repo makes the project clear as day and ultra easy to install.

I'm not saying that the replacement is unsafe or something, it's just that I don't care enough messing with the YouTube layout any more than just using a simple extension that you can review at any time.

Just took a look at the Vorapis extension they seem to promote now, and the extension code is obfuscated (made in a way to be barely understandable by other programmers). The thing with this (CustomTube) extension is that it's open source and clear as day to review. That (Vorapis) project seems to be closed source, and only its devs can do anything with it. It does NOT necessarily mean it's dangerous (it could be for performance), but just worth noting.

I think I'll try forking the CustomTube repo and do stuff with it.

**Reply by MesaBlack** [reply]
NICE!

**Reply by Aleksandra0056** [reply]
Yes!

**Reply by gameshowrecaps** [reply]
If it turns out better and fixes more bugs I'll likely use it. Otherwise, I'll stay here until this breaks majorly. From there, I don't think I'll use StarTube, or go back to Redux. If anything I may have to deal with the new layout using up memory.

**Reply by bangbangyouall** [reply]
just fucking move on gameshowrecaps not everything has to be open source

**Reply by gameshowrecaps** [reply]
I'm not moving on yet, it's not broken enough for my liking for me to do that. When it breaks like fully, I will. There is also no need to be rude.

**Reply by VORAPIS** [reply]
you know that any form of javascript script show everything it does in sources and network debugging menu of browser, right? you arent using v3 because you cant read the code? were you planning on reading all the 70,453 lines of the source code?

**Reply by Aleksandra0056** [reply]
> just fucking move on gameshowrecaps not everything has to be open source

Rude words like the second word in your message starting with "f" are not tolerated on GitHub, so stop being rude. Also, stop attacking to open-source software since nobody likes proprietary due to privacy and telemetry issues.

If you don't like something, then express your opinion politely or leave.

**Reply by Aleksandra0056** [reply]
> If it turns out better and fixes more bugs I'll likely use it. Otherwise, I'll stay here until this breaks majorly. From there, I don't think I'll use StarTube, or go back to Redux. If anything I may have to deal with the new layout using up memory.

By the way, what are StarTube and Redux if I can ask?

**Reply by gameshowrecaps** [reply]
StarTube is the successor to CustomTube, which allows the use of the 2011-2022 layout and is customizable like CustomTube. However it requires V3. Redux is very inaccurate and buggy.

**Reply by Aleksandra0056** [reply]
> StarTube is the successor to CustomTube, which allows the use of the 2011-2022 layout and is customizable like CustomTube. However it requires V3. Redux is very inaccurate and buggy.

I agree with you. CustomTube was much better than StarTube since CustomTube was an independent extension with its own features, but StarTube is dependent on V3 which isn't ideal since independent software is always better than plug-in, script or extension to other software.

I don't even know why lightbeam24 betrayed us by discontinuation of CustomTube. At least lightbeam24 could write a reason for discontinuation and/or fix major issues, but he isn't even responding to his own users and he has been inactive on GitHub since 19. 3. 2024.

**Reply by Aleksandra0056** [reply]
2 dislikes? For what? For my opinion? Such undemocratic and disrespect.

**Reply by lightbeam24** [reply]
> At least lightbeam24 could write a reason for discontinuation and/or fix major issues, but he isn't even responding to his own users and he has been inactive on GitHub since 19. 3. 2024.

As I said in https://github.com/lightbeam24/CustomTube/discussions/240, I had a reason for that, though maybe it wasn't the right choice. Either way, I tried to address everything there.

**Comment by gameshowrecaps** [comment]
I'll stick with CustomTube unless there's a better extension but that's unlikely right now.

**Reply by userunknown358** [reply]
Have you switched to V3 and Startube? If not, download it there: vorapis.pages.dev

**Reply by gameshowrecaps** [reply]
I actually gave it a try once and reverted back to CustomTube.

**Reply by lightbeam24** [reply]
> I actually gave it a try once and reverted back to CustomTube.

Genuinely curious, what made you go back?

**Reply by gameshowrecaps** [reply]
It was a bit too confusing for me.

**Reply by lightbeam24** [reply]
Understandable, that's probably one of StarTube's biggest faults, and I'm looking to mitigate it in the future.

However, you should know that simply applying layout presets is not difficult. All you do is:
1. Click the button on the bottom right corner of the screen
2. Click "StarTube settings"
3. Click one of the preset buttons (ex: Use 2017) and the page will refresh with the new layout

Further customization is more complicated and is the reason why I wasn't gonna mention StarTube here yet, since I do want to fix that eventually.

**Comment by Lort533** [comment]
Anyone still around?
I AM working on a CustomTube fork, but must admit it is *not* the easiest project a person can be involved in.

Here's a random screenshot:
![image](https://github.com/lightbeam24/CustomTube/assets/76491869/b4758992-69f8-4ce1-af8e-f7a4d4fa4546)

As you can see, I did my best to achieve every single bit of nostalgia, leaving nothing much done in a wrong way, making stuff 1:1 like they used to look like.
The issue is stuff that no longer works like it used to. I'll probably have to avoid them at this moment, and slowly add that in future.

Now, my actual questions are, like I said earlier:
- Is anyone still around, interested in this fork?
- Which layout do you use?
- What "annoys" you the most in it?

Second and third questions are important, so I can know on what do I have to focus on and fix on which layout.

Thanks for reading this!

**Reply by Aleksandra0056** [reply]
It looks very interesting and nice. However, I know almost **nothing** about programming in JavaScript or CSS, so I can't help with work.

But I like how that fork looks. By the way, my favourite layouts are 2008, 2009 & 2012.

The most annoying feature for me is infinite scrolling for comments, search results & related videos. And the hidden dislike counter for video comments is an annoying feature too (And the Return YT dislike team is doing nothing with it).

I want also to advice, that you should make a new issue with the title "Hey people, I'm working on a new fork since this looks abandoned" or something else because more people are clicking on the "Issues" section instead of the "Discussions", and this looks so nice, so it deserves more attention.

**Reply by VelvetShadows** [reply]
I'm partial to the 2019 layout myself, but at this point anything would be welcome compared to the current layout.
The biggest aggravating point in current layouts is how enormous everything is, YouTube absolutely despises the concept of content density.  Even something as simple as being able to choose the number of videos per row massively improves YT's navigability.

**Reply by gameshowrecaps** [reply]
There's a bug in the current version with the discussions tab. I use the 2015 layout by the way.
![image](https://github.com/lightbeam24/CustomTube/assets/57842610/15f6a08b-729d-40ce-a457-82b2d598523d)

**Reply by Aleksandra0056** [reply]
Good morning, any news if I can ask?

**Reply by Lort533** [reply]
Thanks for asking.

Indeed I've been working to add more features, however I've got a summer job that will last up until September.
I may work on it in meanwhile, unsure of that, but YouTube has broken a few things in the meanwhile as I saw.
Ironically, the last day of me working on CustomTube was the exact day my (and CustomTube's in general) code for channels have stopped working because YouTube has just pushed the change that broke channels nearly entirely, again.
However, other things (for 2012 layout) that I've fixed like main page, website responsiveness (at least tried) and trending/music/games tabs still do work, so if you are seriously interested, I can publish that before I start working on it again and fixing stuff.

**Reply by gameshowrecaps** [reply]
> Thanks for asking.
> 
> Indeed I've been working to add more features, however I've got a summer job that will last up until September. I may work on it in meanwhile, unsure of that, but YouTube has broken a few things in the meanwhile as I saw. Ironically, the last day of me working on CustomTube was the exact day my (and CustomTube's in general) code for channels have stopped working because YouTube has just pushed the change that broke channels nearly entirely, again. However, other things (for 2012 layout) that I've fixed like main page, website responsiveness (at least tried) and trending/music/games tabs still do work, so if you are seriously interested, I can publish that before I start working on it again and fixing stuff.

Thanks for the update. It's gonna be hard to keep up by then because YouTube's gonna implement changes. I'm definitely not using V3 with StarTube. If CustomTube breaks altogether, I will resort to downloading videos, which thankfully hasn't happened yet.

**Reply by userunknown358** [reply]
Any updates yet?

**Reply by Aleksandra0056** [reply]
> Thanks for asking.
> 
> Indeed I've been working to add more features, however I've got a summer job that will last up until September. I may work on it in meanwhile, unsure of that, but YouTube has broken a few things in the meanwhile as I saw. Ironically, the last day of me working on CustomTube was the exact day my (and CustomTube's in general) code for channels have stopped working because YouTube has just pushed the change that broke channels nearly entirely, again. However, other things (for 2012 layout) that I've fixed like main page, website responsiveness (at least tried) and trending/music/games tabs still do work, so if you are seriously interested, I can publish that before I start working on it again and fixing stuff.

Publishing would be great even if it's still in the Alpha phase since I'm extremely interested in the new version (And many people too).

Could you upload the .xpi file to MediaFire and post a link to us?

**Reply by springboarder06** [reply]
A CustomTube fork is not needed when V3  (and maybe StarTube) exists

**Reply by Aleksandra0056** [reply]
> A CustomTube fork is not needed when V3 (and maybe StarTube) exists

But Lort533 decided, so stop attacking us.

**Reply by iminonet** [reply]
> Anyone still around? I AM working on a CustomTube fork, but must admit it is _not_ the easiest project a person can be involved in.
> 
> Here's a random screenshot: ![image](https://private-user-images.githubusercontent.com/76491869/339881853-b4758992-69f8-4ce1-af8e-f7a4d4fa4546.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI1NTI0OTQsIm5iZiI6MTcyMjU1MjE5NCwicGF0aCI6Ii83NjQ5MTg2OS8zMzk4ODE4NTMtYjQ3NTg5OTItNjlmOC00Y2UxLWFmOGUtZjdhNGQ0ZmE0NTQ2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAxVDIyNDMxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjYjk5YjdkNjgzZDJhYzUzMDVmODVlMjc5MWIyM2IzZjM0ZWQ4ZmYzZDI0ZmQ0YzE2OGZhZDhlMGI2N2Y2ZWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KwnVhcWRh6terTH_PLAOjhZRjjHWBfU5wlPfqcPAONM)
> 
> As you can see, I did my best to achieve every single bit of nostalgia, leaving nothing much done in a wrong way, making stuff 1:1 like they used to look like. The issue is stuff that no longer works like it used to. I'll probably have to avoid them at this moment, and slowly add that in future.
> 
> Now, my actual questions are, like I said earlier:
> 
> * Is anyone still around, interested in this fork?
> * Which layout do you use?
> * What "annoys" you the most in it?
> 
> Second and third questions are important, so I can know on what do I have to focus on and fix on which layout.
> 
> Thanks for reading this!

i hope for the best of u since youtube changes shit in polymer every day sooooo

**Comment by uy813** [comment]
i quit using CustomTube

**Reply by Aleksandra0056** [reply]
Reason?

**Reply by userunknown358** [reply]
> Reason?

They have switched to V3 and V3 is more accurate.

**Reply by Aleksandra0056** [reply]
> > Reason?
> 
> They have switched to V3 and V3 is more accurate.

No, V3 isn't accurate. I loved CustomTube, but the developer (lightbeam24) betrayed me and the whole community because of discontinuation.

Fortunately, Lort533 is working on the CustomTube fork.

**Reply by userunknown358** [reply]
> > > Reason?
> > 
> > 
> > They have switched to V3 and V3 is more accurate.
> 
> No, V3 isn't accurate. I loved CustomTube, but the developer (lightbeam24) betrayed me and the whole community because of discontinuation.
> 
> Fortunately, Lort533 is working on the CustomTube fork.

No disrespect but do you care to explain how? If it's missing features then V3 is still in development. And for everything else, there are userscripts that can increase accuracy of older layouts. Mainly 2016-2017. 

I'm aware that it is not entirely accurate because of the Polymer 2016-2021 layouts.

**Reply by VORAPIS** [reply]
> > > Reason?
> > 
> > 
> > They have switched to V3 and V3 is more accurate.
> 
> No, V3 isn't accurate. I loved CustomTube, but the developer (lightbeam24) betrayed me and the whole community because of discontinuation.
> 
> Fortunately, Lort533 is working on the CustomTube fork.

V3 is literally (technically) 2013 html, how is it not accurate??

**Reply by iminonet** [reply]
> ####

tell me how is it not accurate then compared to customtube :/

**Comment by lightbeam24** [comment]
Abandoned, as explained in fairly great detail here: https://github.com/lightbeam24/CustomTube/discussions/240

**Comment by Lort533** [comment]
Alright, so there was enormous activity here while I couldn't reply, so let me catch up a bit.

CustomTube is now officially abandonded, and like I said earlier, I'll be working back on a fork of this project next month.
Within one month of my inactivity YouTube has already broke random stuff *again*, so yes, it's a hard task to maintain current code flow.

### I guess I have to clarify few things first.

First of all, I **never** told anyone what should they use, whether it's StarTube/V3 or CustomTube. Use whatever you prefer.
I also **never** accused V3 of being malicious, my main point was that it's obfuscated, so it's harder to look through, modify and fix the code if needed. CustomTube is open-source, straightforward extension that doesn't require ANYTHING additional to run.
To anyone who says it's useless for anything else than StarTube/V3 to exist - it's like saying that Linux is useless, because Windows is much easier to work with.

### Anyways, what are the plans for the fork though?

Well, from the very beginning I knew that YouTube keeps changing, and that things keep breaking. I know that relying on YouTube's website structure is pointless, it's a temporary solution. However, changing this behaviour is not a five minute fix - to make CustomTube any independent from breaking, I will have to replace a lot of CustomTube's code. That's why I'm unsure of how exactly I'll be working yet. I may try making things work as intended, and then I would keep fixing them up, while also I would be trying to make CustomTube as independent from YouTube's current HTML and as much "unexpected change" resistant as it can get.

I don't know if there is anything else to say.

**Reply by iminonet** [reply]
Of course the [Aleksandra0056](https://github.com/Aleksandra0056) person reacted with a like

**Reply by Aleksandra0056** [reply]
Mr. Lort533, we should open another place for discussion about your fork (e.g. Reddit, Discord, etc) in case if this project (CustomTube) get eventually archived by lightbeam24 (archiving means the project would be locked permanently) because I have bad feeling that it would eventually happen.

**Reply by iminonet** [reply]
> Mr. Lort533, we should open another place for discussion about your fork (e.g. Reddit, Discord, etc) in case if this project (CustomTube) get eventually archived by lightbeam24 (archiving means the project would be locked permanently) because I have bad feeling that it would eventually happen.

why do you still care about customtube

**Reply by Aleksandra0056** [reply]
> > Mr. Lort533, we should open another place for discussion about your fork (e.g. Reddit, Discord, etc) in case if this project (CustomTube) get eventually archived by lightbeam24 (archiving means the project would be locked permanently) because I have bad feeling that it would eventually happen.
> 
> why do you still care about customtube

Because it was something I liked. Why you are constantly attacking me?

**Reply by Aleksandra0056** [reply]
Hello, any news Mr. Lort533?

**Reply by Lort533** [reply]
Hello,
sorry for the late reply. While I didn't have enough time to work on CustomTube (remake, prob won't preserve the project as is) so far, I've sure made a Discord server back when requested: https://discord.gg/VbSmjaTFB7. It will be way easier to talk to each other in there, and I have quite huge plans on this addon, which, of course, will be open source, and based on open source.

**Comment by uy813** [comment]
I'm afraid you came too late. we can talk each other in v3 server
